flink-libraries/flink-table/src/main/scala/org/apache/flink/table/codegen/MatchCodeGenerator.scala

```diff
@@ -445,4 +498,223 @@ class MatchCodeGenerator(
 
     generateFieldAccess(patternVariableRef.copy(code = NO_CODE), fieldRef.getIndex)
   }
+
+  class AggBuilder(variable: String) {
+
+    private val aggregates = new mutable.ListBuffer[RexCall]()
+
+    private val variableUID = newName("variable")
+
+    private val resultRowTerm = newName(s"aggRow_$variableUID")
+
+    private val rowTypeTerm = "org.apache.flink.types.Row"
+
+    def getOrAddAggregation(call: RexCall): GeneratedExpression = {
+      reusableInputUnboxingExprs.get((call.toString, 0)) match  {
+        case Some(expr) =>
+          expr
+
+        case None =>
+          val exp: GeneratedExpression = generateAggAccess(call)
+          aggregates += call
+          reusableInputUnboxingExprs((call.toString, 0)) = exp
+          exp.copy(code = NO_CODE)
+      }
+    }
+
+    private def generateAggAccess(call: RexCall) = {
+      val singleResultTerm = newName("result")
+      val singleResultNullTerm = newName("nullTerm")
+      val singleResultType = FlinkTypeFactory.toTypeInfo(call.`type`)
+      val singleResultTypeTerm = boxedTypeTermForTypeInfo(singleResultType)
+
+      val patternName = findEventsByPatternName(variable)
+
+      val codeForAgg =
+        j"""
+           |$rowTypeTerm $resultRowTerm = calculateAgg_$variableUID(${patternName.resultTerm});
+           |""".stripMargin
+
+      reusablePerRecordStatements += codeForAgg
+
+      val defaultValue = primitiveDefaultValue(singleResultType)
+      val codeForSingleAgg =
+        j"""
+           |boolean $singleResultNullTerm;
+           |$singleResultTypeTerm $singleResultTerm = ($singleResultTypeTerm) $resultRowTerm
```

Comment:
Something is wrong here. You are casting a `Row` into the result type of the aggregation?

Reply:
Hmmm, no I am not. Have you maybe missed the `.getField()`?

Reply:
Sorry about this comment I totelly missed the following line.